### DISCUSSION TITLE
Payment method

### DISCUSSION THREAD

**Discussion by aerl215** [n3d1117/chatgpt-telegram-bot#116] [Q&A]
Hi, I'm new to programming and in the process of company in your code I got interested in python language. I just started learning it and want to clarify, does your code already have a pay per request feature or not?  As I understand this code has tracking for each user tokens spent, is it possible to add pay per use and maybe you can give me a hint or hint how this can be implemented?

**Comment by aerl215** [comment]
And I'm grateful to you for sharing your work, thanks to your work I am seriously interested in development. I hope my question is not too brazen and vulgar. You are very cool).

**Comment by n3d1117** [comment]
Hey there @aerl215! Can you elaborate on what do you mean by "pay per request"? As in integrated payments within the bot?

**Reply by aerl215** [reply]
Yes, I meant to integrate payments inside the bot, as I understand your code already has a Venice of counting tokens spent by users

**Reply by n3d1117** [reply]
@aerl215 You can check out https://core.telegram.org/bots/payments and https://docs.python-telegram-bot.org/en/stable/examples.paymentbot.html as starting points! Good luck

**Comment by AlexHTW** [comment]
Hi @aerl215,
I think a per use payment (meaning a user sends a prompt and receives an answer and invoice for that) system is not feasible and user-friendly. Especially because you don't know how expensive a request will be.
The bot implements monthly* budgets that you could sell through telegram payments. For that you would need to add a bot-command that sends the invoice, processes the checkout and updates the budget value in your .env file for that user.

*the monthly budget in the current implementation is not ideal for selling, because it resets on the first of each month. This will probably make you or your users unhappy. However I suggest a few additions for buget settings [here](https://github.com/n3d1117/chatgpt-telegram-bot/issues/106#issuecomment-1485718002) that I will try to work on soon. Then it would be more straightforward to sell unrestricted budget which will not reset monthly.

**Reply by chxmontoro** [reply]
@AlexHTW Is it possible to assist with the implementation of monthly subscription payment? For example, i want to charge $5/month to use the bot. I will be willing to contribute to make this possible.

**Reply by AlexHTW** [reply]
Hey @chxmontoro, payments is a goal I had in mind from the beginning with the budgets. However, I have no experience with telegram payments and currently not much time to learn it or work on this project at all. So I can't really offer much assistance in that direction right now, sorry. 
If my circumstances change I will gladly message you to collaborate on this feature.

**Reply by chxmontoro** [reply]
Bumping on this. Any solution to this @AlexHTW ?

**Reply by AlexHTW** [reply]
> Bumping on this. Any solution to this @AlexHTW ?

Hey! No, sadly no time to work on this project currently.
